unix/vncserver/vncsession.c

```diff
@@ -546,7 +547,8 @@ run_script(const char *username, const char *display, char **envp)
     switch_user(pwent->pw_name, pwent->pw_uid, pwent->pw_gid);
 
     if (chdir(pwent->pw_dir) == -1)
-        chdir("/");
+        if (chdir("/") == -1)
+            syslog(LOG_CRIT, "chdir: %s", strerror(errno));
```

Comment:
The second `chdir()` is an emergency fallback. So we should probably log the issue with the first one in that case.

And a more descriptive error message so the user/sysadmin knows what failed.